src/na/na_types.h

```diff
@@ -135,10 +136,18 @@ typedef int (*na_cb_t)(const struct na_cb_info *callback_info);
 #define NA_NO_BLOCK 0x01 /*!< no blocking progress */
 #define NA_NO_RETRY 0x02 /*!< no retry of operations in progress */
 
+/* Thread modes (default is thread-safe) */
+#define NA_THREAD_MODE_SINGLE_CLS                                              \
+    (0x01) /*!< only one thread will access class */
+#define NA_THREAD_MODE_SINGLE_CTX                                              \
+    (0x02) /*!< only one thread will access context */
+#define NA_THREAD_MODE_SINGLE                                                  \
+    (NA_THREAD_MODE_SINGLE_CLS | NA_THREAD_MODE_SINGLE_CTX)
+
```

Comment:
It seems like the number of threads is not the important thing. Rather, it's important that if more than one thread uses the same class or context, that their access is serialized? So _SERIAL rather than _SINGLE here?

Reply:
Yes that's true, I think I like that change to serial